jwt.go

```diff
@@ -62,10 +69,70 @@ func (s *SessionClaims) HasRole(role string) bool {
 	return s.ActiveOrganizationRole == role
 }
 
-// Claims holds generic JWT claims.
+// RegisteredClaims holds public claim values (as specified in RFC 7519).
+type RegisteredClaims struct {
```

Comment:
❓ Instead of retyping/copying the struct, what about create an alias instead?
```golang
type RegisteredClaims jwt.Claims
```

Reply:
I'm not sure if it's better to add a type alias or break away completely from go-jose and redefine everything on our own.

Adding a type alias means that our type can be cast to the go-jose type and this way we can use all of go-jose's methods. Is this something we want? Could be a benefit or not.

It also enables us to take advantage of json unmarshaling without overriding any methods. This is clearly a benefit. The less code we have to write, the better.

However, if we redefine everything ourselves, we completely hide go-jose. There's no way you can use the unexported go-jose field and there's no need to ever reach for it.

I'm happy to change this if you think it's better. I don't have a strong opinion. :smile:

Reply:
If you define a type alias I believe someone can still do `jwt.Claims(claims)` thus potentially going against the encapsulation we're trying to achieve here. I think embedding/unexporting the jose struct might be inevitable.

Reply:
Thanks for the explanation, makes sense to me!